src/components/map/LayersService.js

```diff
@@ -649,7 +729,107 @@ goog.require('ga_urlutils_service');
+                  maxZoom: config.maxZoom,
+                  url: config.url
+                })
+              });
+            }
+            if (config.sourceId && config.styleUrl) {
+              var sourceId = config.sourceId;
+              $http.get(config.styleUrl, {
+                cache: true
+              }).then(function(response) {
+                var glStyle = response.data;
+
+                if (config.sourceType !== 'raster') {
+
+                  // HACK: Make the Swiss style transparent to see the relief
+                  glStyle.layers.forEach(function(style, idx) {
+                    if (style.id === 'Schweiz') {
+                      style.paint['fill-opacity'] = 0.8;
+                    }
+                  });
+
+                  // For vector source we parse then apply the style after
+                  // loading the sprite image.
+                  var spriteConfigUrl = glStyle.sprite + '.json';
+
+                  // Load the style for a specific source
+                  $http.get(spriteConfigUrl, {
+                    cache: true
+                  }).then(function(response) {
+                    $window.olms.stylefunction(olLayer, glStyle,
+                        config.sourceId,
+                        undefined, response.data, glStyle.sprite + '.png',
+                        ['Helvetica']);
+                  }, function(reason) {
+                    $window.console.error('Apply style failed. Reason: "' +
+                        reason + '"');
+                  });
+                } else {
+                  glStyle.layers.forEach(function(style, idx) {
+                    if (style.source === config.sourceId) {
+                      // TODO: We should be able to parse the style for raster.
+                      // Verify if it exists or not in mb2olstyle library
+                    }
+                  });
+                }
+
+                // Load the tileset config
+                if (!config.url) {
+                  var sourceConfigUrl = glStyle.sources[sourceId].url;
+                  $http.get(sourceConfigUrl, {
+                    cache: true
+                  }).then(function(response) {
+                    var data = response.data;
+                    var olSource;
+                    if (config.sourceType === 'raster') {
+                      olSource = new ol.source.XYZ({
+                        minZoom: data.minZoom,
+                        maxZoom: data.maxZoom,
+                        urls: data.tiles
+                      });
+                      olLayer.setOpacity(config.opacity || 1);
+                    } else { // vector
+                      olSource = new ol.source.VectorTile({
+                        format: new ol.format.MVT(),
+                        minZoom: data.minzoom,
+                        maxZoom: data.maxzoom,
+                        urls: data.tiles
+                      });
+                    }
```

Comment:
I think we should use `ol.tile.TileJSON` and a dedicated service for loading/editing the glstyle.

see: https://openlayers.org/en/latest/apidoc/module-ol_source_TileJSON.html